### PR TITLE
Swallow and log parsing exceptions on immutable_metadata in AWS

### PR DESCRIPTION
**Pull request by gardleopard**
One subnet with broken json in immutable_metadata tag can make the entire `subnets/aws` call to clouddriver fail with a 500 error.
This PR mitigates this by swallowing the exception and logging the message in the ERROR log.

```
curl http://clouddriver.url/subnets/aws
{
  "error" : "Internal Server Error",
  "exception" : "com.fasterxml.jackson.core.JsonParseException",
  "message" : "Unexpected character ('=' (code 61)): was expecting a colon to separate field name and value\n at [Source: {\"purpose\": \"internal-foobar\",\"target\"=\"ec2\"}; line: 1, column: 48]",
  "status" : 500,
  "timestamp" : 1488549571234
}
```

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy [hunk 1]
```diff
@@ -74,10 +77,14 @@ class AmazonSubnetProvider implements SubnetProvider<AmazonSubnet> {
     String target = null
     if (json) {
       def objectMapper = new ObjectMapper()
-      def metadata = objectMapper.readValue((String) json, Map.class)
-      metadata?.purpose
-      purpose = metadata?.purpose
-      target = metadata?.target
+      try {
+        def metadata = objectMapper.readValue((String) json, Map.class)
+        metadata?.purpose
```
Comment (anotherchrisberry) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy]: This line seems really odd (I know it's in the original code, just doesn't seem to serve any purpose). Can we remove it?

Comment (gardleopard) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonSubnetProvider.groovy]: Yeah, I will remove it.

**Comment by gardleopard** [comment]
We discovered this when users reported that they could not set up deployments. The page for creating got stuck with the loading animation while getting 429 from the subnet endpoint in gate.

**Comment by gardleopard** [comment]
Added more debug friendly log output:
```
10:44:25.067 [Test worker] ERROR com.netflix.spinnaker.clouddriver.aws.provider.view.AmazonSubnetProvider - Can not extract purpose and/or target from immutable_metadata
	Account: test
	Subnet id: subnet-00000001
	Exception message: Unexpected character ('=' (code 61)): was expecting a colon to separate field name and value
 at [Source: {"JSON"="BROKEN","target": "EC2"}; line: 1, column: 9]
```

**Comment by gardleopard** [comment]
@spinnaker/reviewers PTAL

**Comment by gardleopard** [comment]
@spinnaker/reviewers PTAL

**Comment by anotherchrisberry** [comment]
LGTM

**Pull request merged**